Subject: [PATCH] [BZ 184396] zipl: Add missing options to help output
From: Stefan Haberland <sth@linux.ibm.com>

Description:   zipl: fix secure boot config handling
Symptom:       The config file parsing for secure boot worked not as
               it was expected to be. For example a config section
               setting was not evaluated properly.
               It is not possible to specify command line option -S
               without other options.
               Additionally the man page showed an invalid example.
Problem:       The config file parsing was not implemented properly.
Solution:      The hierarchy of the secure boot settings in the config
               file is:
                    defaultboot > menu > section
               Allow that --secure or -S is specified on command line
               without the need to allow all options on the command
               line. Also ensure that the command line option
               overrules the config option and correctly ensure that
               secure boot is only set for SCSI devices.
               Fix man page example.
Reproduction:  Run zipl with a secure= setting in a configuration
               section or specify -S on command line.
Upstream-ID:   dcce14923c3e9615df53773d1d8a3a22cbb23b96
Problem-ID:    184396

Upstream-Description:

              zipl: Add missing options to help output

              Reviewed-by: Stefan Haberland <sth@linux.ibm.com>
              Signed-off-by: Jan Hoeppner <hoeppner@linux.ibm.com>


Signed-off-by: Stefan Haberland <sth@linux.ibm.com>
---
 zipl/src/zipl.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- a/zipl/src/zipl.c
+++ b/zipl/src/zipl.c
@@ -68,6 +68,8 @@ static const char* usage_text[] = {
 "-P, --parameters PARMLINE       Use specified kernel PARMLINE",
 "-T, --tape TAPEDEV              Install bootloader on tape device TAPEDEV",
 "-s, --segment SEGMENT,ADDR      Install a segment from file SEGMENT",
+"-k, --kdump=auto                Install a kdump kernel that can be used as a",
+"                                stand-alone dump tool",
 "-d, --dumpto DUMPDEV[,SIZE]     Install a system dump record on tape device",
 "                                or disk partition DUMPDEV",
 "-M, --mvdump DEVLIST[,SIZE]     Install a multi-volume dump record on each",
@@ -78,7 +80,12 @@ static const char* usage_text[] = {
 "-n, --noninteractive            Answer all confirmation questions with 'yes'",
 "-V, --verbose                   Provide more verbose output",
 "-a, --add-files                 Add all referenced files to bootmap file",
-"    --dry-run                   Simulate run but don't modify IPL records"
+"    --dry-run                   Simulate run but don't modify IPL records",
+"-S, --secure SWITCH             Control the zIPL secure boot support.",
+"                                 auto (default):",
+"                                  Write signatures if available and supported",
+"                                 1: Write signatures regardless of support",
+"                                 0: Do not write signatures"
 };
 
 
